optimum/intel/openvino/quantization.py

```diff
@@ -784,47 +797,67 @@ def disable_progress_bar(self, disable: bool = True) -> None:
         else:
             self.model._progress_bar_config["disable"] = disable
 
-    def _prepare_feature_extraction_calibration_data(
+    def _prepare_text_encoder_model_calibration_data(
         self,
         quantization_config: OVQuantizationConfigBase,
         dataset: "Dataset",
         seq_len: int = 128,
     ) -> OVCalibrationDataset:
-        from optimum.intel import OVModelForFeatureExtraction, OVSentenceTransformer
+        """
+        Prepares calibration data for text-encoder-like models.
+        Supports OVModelForFeatureExtraction, OVModelForMaskedLM and OVSentenceTransformer.
+        """
+        from optimum.intel import OVModelForFeatureExtraction, OVModelForMaskedLM, OVSentenceTransformer
 
         self.model.compile()
 
         num_samples = quantization_config.num_samples or 128
         calibration_data = []
         try:
+            inference_result_mock = {}
+            if isinstance(self.model, OVModelForFeatureExtraction):
+                inference_result_mock["last_hidden_state"] = np.empty((1,), np.float32)
+            elif isinstance(self.model, OVModelForMaskedLM):
+                inference_result_mock["logits"] = np.empty((1,), np.float32)
+            elif isinstance(self.model, OVSentenceTransformer):
+                inference_result_mock["token_embeddings"] = np.empty((1,), np.float32)
+                inference_result_mock["sentence_embedding"] = np.empty((1,), np.float32)
+            else:
+                raise RuntimeError(
+                    f"Unsupported model type {type(self.model).__name__} for calibration dataset collection."
+                )
+
             self.model.request = InferRequestWrapper(
                 self.model.request,
                 calibration_data,
-                inference_result_mock={
-                    "last_hidden_state": np.empty((1,), np.float32),
-                    "token_embeddings": np.empty((1,), np.float32),
-                    "sentence_embedding": np.empty((1,), np.float32),
-                },
+                inference_result_mock=inference_result_mock,
             )
 
-            if isinstance(self.model, OVModelForFeatureExtraction):
-                tokenizer = AutoTokenizer.from_pretrained(
-                    quantization_config.tokenizer, trust_remote_code=quantization_config.trust_remote_code
-                )
-            elif isinstance(self.model, OVSentenceTransformer):
+            if isinstance(self.model, OVSentenceTransformer):
                 tokenizer = self.model.tokenizer
             else:
-                raise RuntimeError(
-                    f"Unsupported model type {type(self.model).__name__} for calibration dataset collection."
+                tokenizer = AutoTokenizer.from_pretrained(
+                    quantization_config.tokenizer, trust_remote_code=quantization_config.trust_remote_code
                 )
 
+            seq_len = min(seq_len, self.model.config.max_position_embeddings or seq_len)
```

Comment:
wouldn't this result in issues for cases where `self.model.config.max_position_embeddings` has a value like -1 ? https://github.com/huggingface/transformers/blob/eefc86aa316a361bbb232eb6b547ce86e185695a/src/transformers/models/xlnet/configuration_xlnet.py#L230

Reply:
Thanks!